File: PDP_Odev2/KISIM1/Odev2Kutuphane/src/RASTGELEKISIURET/RastgeleKisi.java
/**
*
* @author Sedanur Sarıkoç(sedanur.sarikoc@ogr.sakarya.edu.tr), Merve Sarı(merve.sari3@ogr.sakarya.edu.tr)
* @since 13.04.2020
* <p>
* Kişi bilgilerini derleyerek rastgele kişi oluşturan sınıf
* </p>
*/
package RASTGELEKISIURET;


public class RastgeleKisi {
    
    private String RastgeleKisi = "";
    
    Kisi kisi = new Kisi();
    
    public RastgeleKisi(){
        
        RastgeleKisi += kisi.KimlikNoDondur(); 
        RastgeleKisi += " ";
        
        RastgeleKisi += kisi.IsimDondur();
        RastgeleKisi += " ";
        
        RastgeleKisi += kisi.YasDondur();
        RastgeleKisi += " ";
        
        RastgeleKisi += kisi.TelefonDondur();
        RastgeleKisi += " (";
        
        RastgeleKisi += kisi.IMEINoDondur();
        RastgeleKisi += ")";
        
    }
    
    public String RastgeleKisiOlustur(){

        return RastgeleKisi;
    }
}

File: PDP_Odev2/KISIM2/src/RastgeleKisi.c
/**
*
* @author Sedanur Sarıkoç(sedanur.sarikoc@ogr.sakarya.edu.tr), Merve Sarı(merve.sari3@ogr.sakarya.edu.tr)
* @since 20.04.2020
* <p>
* Kişi bilgilerini derleyerek rastgele kişi oluşturan kaynak dosyası
* </p>
*/
#include "Kisi.h"
#include "RastgeleKisi.h"
#include "stdio.h"
#include "stdlib.h"
#include "unistd.h"


RastgeleKisi RastgeleKisiOlustur(){
	
	RastgeleKisi this;
	this = (RastgeleKisi)malloc(sizeof(struct RASTGELEKISI));
	this->RastgeleKisiBilgileri = (char*)malloc(sizeof(char)*1000);
	
	this->rastgelekisi = &RastgeleKisiDondur;
    this->yoket = &RastgeleKisiYoket;
		
	return this;
}

char* RastgeleKisiDondur(const RastgeleKisi this){  
	Kisi kisi = KisiOlustur();
	sprintf(&this->RastgeleKisiBilgileri[0],"%s %s %d %s %s%s%s", kisi->kimlik(kisi), kisi->isim(kisi), kisi->yas(kisi), kisi->telefon(kisi), "(", kisi->imei(kisi), ")");
	kisi->yoket(kisi);
	return this->RastgeleKisiBilgileri; 
}

void RastgeleKisiYoket(RastgeleKisi this){
    if(this == NULL) return;
	this->RastgeleKisiBilgileri = NULL;
	free(this->RastgeleKisiBilgileri);
    free(this);
}
